presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/PlanChecker.java

```diff
@@ -67,36 +80,88 @@ public PlanChecker(FeaturesConfig featuresConfig, boolean forceSingleNode)
                         new VerifyNoIntermediateFormExpression(),
                         new VerifyProjectionLocality(),
                         new DynamicFiltersChecker(),
-                        new WarnOnScanWithoutPartitionPredicate(featuresConfig));
+                        new WarnOnScanWithoutPartitionPredicate(featuresConfig)));
         if (featuresConfig.isNativeExecutionEnabled() && (featuresConfig.isDisableTimeStampWithTimeZoneForNative() ||
                 featuresConfig.isDisableIPAddressForNative())) {
-            builder.put(Stage.INTERMEDIATE, new CheckUnsupportedPrestissimoTypes(featuresConfig));
+            checkers.put(Stage.INTERMEDIATE, new CheckUnsupportedPrestissimoTypes(featuresConfig));
         }
-        checkers = builder.build();
     }
 
-    public void validateFinalPlan(PlanNode planNode, Session session, Metadata metadata, WarningCollector warningCollector)
+    public synchronized void update(PlanCheckerProvider provider)
```

Comment:
@tdcmeehan comments from https://github.com/prestodb/presto/pull/23596/commits/486737b438220a1d4fa954e8a054e457bf77ad96#r1765240968

> Let's do the following:
>1. Let's make checkers volatile. This will be accessed in a multithreaded environment, and the volatile keyword will ensure visibility after updates across multiple threads.
>2. Let's remove providedInitialized, and instead add a method update which takes in a PlanCheckerProvider, which is called by the PlanCheckerProviderManager once the manager is initialized.
>3. Now that the flow of communication is reversed, we don't need ensureProvidedInitialized.
>4. Add a @GuardedBy("this") to the class to document the synchronization of this class.

My initial thoughts against this were because it leads to being able to create a new arbitrary `PlanChecker` or `PlanFragmenter` that could easily not be updated with the plugin provider. It's also where and how to call `update`, since you need to get an instance of the `PlanCheckerProviderManager` first. The changes in `LocalQueryRunner` show these extra steps.